### ISSUE TITLE
False positive in SleepInsteadOfDelay rule in 1.23.0

### ISSUE THREAD

**Issue by mannodermaus** [detekt/detekt#6176]
## Expected Behavior

The code block below should not trigger Detekt's `DelayInsteadOfSleep` rule.

## Observed Behavior

After updating to 1.23.0, it does.

## Steps to Reproduce

The code block below, albeit convoluted, triggers a false-positive of the `SleepInsteadOfDelay` rule. The method is not suspending, nor is it ever called from within a coroutine.

```kotlin
class OhBoy {
    private val lock = Object()

    fun get(a: Int, b: Int) {
        synchronized(lock) {
            if (a == b) {
                var counter = 3

                do {
                    try {
                        return
                    } catch (e: IllegalArgumentException) {
                        e.printStackTrace()
                        Thread.sleep(500)
                    }
                } while (counter-- > 0)

            } else {
                println("else")
            }
        }
    }
}
```

It's likely the number of nested blocks that may trip up Detekt here. After removing the `synchronized` block and the `do-while` loop, the rule is not triggered.

## Context

It's a piece of code in an older area of the codebase, which was suddenly brought to our attention again after the update to Detekt 1.23.0. There is no involvement of coroutines anywhere near this call site.

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): 8.1.1
* Gradle scan link (add `--scan` option when running the gradle task): -/-
* Operating System and version: macOS 13.3.1
* Link to your project (if it's a public repository): -/-
